docs/proposal.md

```diff
@@ -0,0 +1,108 @@
+Containerd CRI Integration
+=============
+Author: Lantao Liu (@random-liu)
+## Abstract
+This proposal aims to integrate [containerd](https://github.com/containerd/containerd) with Kubelet against the [container runtime interface (CRI)](https://github.com/kubernetes/kubernetes/blob/v1.6.0/pkg/kubelet/api/v1alpha1/runtime/api.proto).
+## Background
+Containerd is a core container runtime, which provides the minimum set of functionalities to manage the complete container lifecycle of its host system, including container execution and supervision, image distribution and storage, etc.
+
+Containerd was previously part of Docker, [introduced in Docker 1.11](https://blog.docker.com/2016/04/docker-engine-1-11-runc/), used to manage [runC](https://runc.io/) containers on the node. As is shown below, it creates a containerd-shim for each container, and the shim manages the lifecycle of its corresponding container.
+![containerd](./containerd.png)
+
+In Dec. 2016, Docker spun it out into a standalone component, and donated it to [CNCF](https://www.cncf.io/) in Mar. 2017.
+
+## Motivation
+Containerd is a potential alternative to Docker as the runtime for Kubernetes cluster. *Compared with Docker*, containerd has pros and cons.
+### Pros
+* **Stability**: Containerd has limited scope and slower feature velocity, which is expected to be more stable.
+* **Compatibility**: The scope of containerd meets Kubernetes' requirement pretty well.
+* **Performance**:
+  * Containerd consumes less resource because it's only a subset of Docker;
+  * Containerd CRI integration eliminates an extra hop in the stack (as is shown below). ![performance](./performance.png)
+* **Neutral Foundation**: Containerd is part of CNCF now.
+### Cons
+* **User Adoption**:
+  * Ideally, Kubernetes users don't interact with the underlying container runtime directly. However, for the lack of debug toolkits, sometimes users still need to login the node to debug with Docker CLI directly.
+  * Containerd also provides barebone CLIs [ctr](https://github.com/containerd/containerd/tree/master/cmd/ctr) and [dist](https://github.com/containerd/containerd/tree/master/cmd/dist) for development and debugging purpose, but it needs time to educate users.
+* **Maturity**: The rescoped containerd is pretty new, and it's still under heavy development.
+## Goals
+* Make sure containerd constantly meet the requirement of Kubernetes.
```

Comment:
I want to address that we want containerd to always meet the requirement, now and the future.